app/main.py

```diff
@@ -1 +1,58 @@
-#  write your code here
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> any:
+        if self.is_hungry is True:
+            self.is_hungry = not self.is_hungry
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(
+            self,
+            name: str,
+            appetite: int = 3,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(name, is_hungry)
+        self.appetite = appetite
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(
+            self,
+            name: str,
+            appetite: int = 7,
+            is_hungry: bool = True) -> None:
+        super().__init__(name, is_hungry)
+        self.appetite = appetite
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    return sum(
+        animal.feed()
+        for animal in animals
+        if isinstance(animal, Animal)
+        and animal.is_hungry is True
```

Comment:
There is no need to perform checks that go after `if` because you initially expect to obtain a list of `Animal` instances and the logic that checks whether the animal is hungry is placed inside of the `.feed()` method.